sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -278,13 +278,49 @@ private[sql] object QueryExecutionErrors extends QueryErrorsBase with ExecutionE
   }
 
   def ansiDateTimeError(e: Exception): SparkDateTimeException = {
+    def extractDateTimeErrorInfo(e: Exception): (String, String, String) = {
+      val errorMessage = e.getMessage
+
+      val valuePattern = "Invalid value for ([A-Za-z]+) \\(valid values (.+)\\): (.+)".r
+      val datePattern = "Invalid date '[A-Z]+ ([0-9]+)'".r
+
+      errorMessage match {
```

Comment:
@srielau If I understood @MaxGekk we want to get away from this message creation in the specific calls of the error, but actually keep all the info in error-conditions.json. This will make it really hard to sync with other APIs, especially as we are expanding these endless number of exception messages that we can get from the java library. Also, since this is external dependancy, who is to guarantee that this message format will stay the same when java updates.